specifyweb/frontend/js_src/lib/components/Statistics/hooks.tsx

```diff
@@ -0,0 +1,424 @@
+import { SpQuery, SpQueryField, Tables } from '../DataModel/types';
+import type { IR, RA, WritableArray } from '../../utils/types';
+import { useMultipleAsyncState } from '../../hooks/useAsyncState';
+import React from 'react';
+import { ajax } from '../../utils/ajax';
+import { statsSpec } from './StatsSpec';
+import type {
+  BackEndStat,
+  BackendStatsResult,
+  CustomStat,
+  DefaultStat,
+  QueryBuilderStat,
+  StatCategoryReturn,
+  StatLayout,
+  StatSpecCalculated,
+  StatsSpec,
+} from './types';
+import { SpecifyResource } from '../DataModel/legacyTypes';
+import { serializeResource } from '../DataModel/helpers';
+import { formatNumber } from '../Atoms/Internationalization';
+import { deserializeResource } from '../../hooks/resource';
+import { addMissingFields } from '../DataModel/addMissingFields';
+import { schema } from '../DataModel/schema';
+import { makeQueryField } from '../QueryBuilder/fromTree';
+import { keysToLowerCase } from '../../utils/utils';
+import { statsText } from '../../localization/stats';
+import { throttledAjax } from '../../utils/ajax/throttledAjax';
+import { unknownCategories, urlSpec } from './definitions';
+import { SerializedResource } from '../DataModel/helperTypes';
+
+/**
+ * Fetch backend statistics from the API
+ */
+export function useBackendApi(
+  categoryToFetch: RA<string>,
+  showDialog = false
```

Comment:
Inconsistent naming. This variable is called `loadingScreen` in useMultipleAsyncState